libraries/ms-common/src/stm32f0xx/gpio.c

```diff
@@ -60,7 +60,7 @@ StatusCode gpio_init_pin(GPIOAddress *address, GPIOSettings *settings) {
 
   // These are default values which are not intended to be changed.
   init_struct.GPIO_Speed = GPIO_Speed_Level_3;
-  init_struct.GPIO_OType = GPIO_OType_PP;
+  init_struct.GPIO_OType = (settings->direction == GPIO_DIR_OUT_OD) ? GPIO_OType_OD : GPIO_OType_PP;
```

Comment:
I'll deal with it when I do my ms-common library cleanup PR once I'm done finals. I plan to revisit interfaces and improve documentation prior to the influx of new members in fall.